packages/desktop-client/src/components/NotesButton.tsx

```diff
@@ -1,116 +1,87 @@
-// @ts-strict-ignore
-import React, { useState } from 'react';
+import React, { useEffect, useRef, useState, type ComponentProps } from 'react';
+import { Popover } from 'react-aria-components';
 
 import { useLiveQuery } from 'loot-core/src/client/query-hooks';
 import { send } from 'loot-core/src/platform/client/fetch';
 import { q } from 'loot-core/src/shared/query';
+import { type NoteEntity } from 'loot-core/types/models';
 
 import { SvgCustomNotesPaper } from '../icons/v2';
-import { type CSSProperties, theme } from '../style';
+import { type CSSProperties, styles, theme } from '../style';
 
 import { Button } from './common/Button';
+import { Tooltip } from './common/Tooltip';
 import { View } from './common/View';
 import { Notes } from './Notes';
-import { Tooltip, type TooltipPosition, useTooltip } from './tooltips';
-
-type NotesTooltipProps = {
-  editable?: boolean;
-  defaultNotes?: string;
-  position?: TooltipPosition;
-  onClose?: (notes: string) => void;
-};
-function NotesTooltip({
-  editable,
-  defaultNotes,
-  position = 'bottom-left',
-  onClose,
-}: NotesTooltipProps) {
-  const [notes, setNotes] = useState<string>(defaultNotes);
-  return (
-    <Tooltip position={position} onClose={() => onClose(notes)}>
-      <Notes
-        notes={notes}
-        editable={editable}
-        focused={editable}
-        onChange={setNotes}
-      />
-    </Tooltip>
-  );
-}
 
 type NotesButtonProps = {
   id: string;
   width?: number;
   height?: number;
   defaultColor?: string;
-  tooltipPosition?: TooltipPosition;
+  tooltipPosition?: ComponentProps<typeof Tooltip>['placement'];
   style?: CSSProperties;
 };
 export function NotesButton({
   id,
   width = 12,
   height = 12,
   defaultColor = theme.buttonNormalText,
-  tooltipPosition,
+  tooltipPosition = 'bottom start',
   style,
 }: NotesButtonProps) {
-  const [hover, setHover] = useState(false);
-  const tooltip = useTooltip();
-  const data = useLiveQuery(() => q('notes').filter({ id }).select('*'), [id]);
-  const note = data && data.length > 0 ? data[0].note : null;
+  const triggerRef = useRef(null);
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const data = useLiveQuery<NoteEntity[]>(
```

Comment:
This would be nice to have as a hook, but that can be done in a separate PR

Reply:
Definitely agreed. But lets try to keep the scope of changes small (i.e. this PR).